bot.py

```diff
@@ -485,7 +507,52 @@ def cache_response(message_pointer, response_ts):
     return None
 
 
+def init_oauth():
```

Comment:
Wondering if the init_oauth() method can be removed from bot and used from IAM_Token class.  It would be easier to maintain since there is duplicate code here now.  The IAM_Token.refresh function does similar thing as the init_oauth() method.